src/ol/layer/group.js

```diff
@@ -221,6 +221,9 @@ ol.layer.Group.prototype.getLayerStatesArray = function(opt_states) {
         layerState.extent = ownLayerState.extent;
       }
     }
+    if (layerState.zIndex === 0 && ownLayerState.zIndex !== 0) {
```

Comment:
I've changed the `zIndex` initialization to allow `undefined` and changed the conditional:
```javascript
if (layerState.zIndex === undefined) {
      if (ownLayerState.zIndex !== undefined) {
        layerState.zIndex = ownLayerState.zIndex;
        layerState.layer.setZIndex(ownLayerState.zIndex);
      } else {
        layerState.zIndex = 0;
        ownLayerState.zIndex = 0;
        layerState.layer.setZIndex(0);
        ownLayerState.layer.setZIndex(0);
      }
    }
```
I also made changes to [`ol.renderer.Map.sortByZIndex`](https://github.com/openlayers/openlayers/blob/master/src/ol/renderer/map.js#L333) to order `zIndex: undefined` the same as `zIndex: 0`.  However, many existing tests expect a layer to initialize with a default `zIndex` of `0`.   Should I go ahead and change these tests to expect `zIndex: 0`?